launch_ros/launch_ros/actions/node.py

```diff
@@ -61,15 +61,18 @@
 class Node(ExecuteProcess):
     """Action that executes a ROS node."""
 
+    UNSPECIFIED_NODE_NAME = '<node_name_unspecified>'
+    UNSPECIFIED_NODE_NAMESPACE = '<node_namespace_unspecified>'
+
     def __init__(
         self, *,
         executable: Optional[SomeSubstitutionsType] = None,
         node_executable: Optional[SomeSubstitutionsType] = None,
         package: Optional[SomeSubstitutionsType] = None,
         name: Optional[SomeSubstitutionsType] = None,
-        namespace: Optional[SomeSubstitutionsType] = '',
+        namespace: Optional[SomeSubstitutionsType] = None,
         node_name: Optional[SomeSubstitutionsType] = None,
-        node_namespace: SomeSubstitutionsType = '',
+        node_namespace: SomeSubstitutionsType = None,
```

Comment:
is this change an API break?

Reply:
To me more clear, changing the default isn't the problem.
The problem is that `namespace=''` meant that the namespace in code was going to be used (no remapping rule passed), and now it means that the namespace is remapped to `/` (`__ns:=/` rule passed).

I think that is a (subtle) API break.
I think that the behavior now is the correct one, but I'm not sure if changing the behavior in a previous distro is ok.

Reply:
Thanks for the review. I've amended the logic so that we don't change the behavior in Foxy (e7e08a7). Now `namespace=None` and `namespace=''` act the same (ie. no remapping rule is passed).